Enable monitoring on a namespace

ipolonsk opened the pull request:
Part of: [operate-first/SRE#216](https://github.com/operate-first/SRE/issues/216 )
fix: https://github.com/operate-first/SRE/issues/277 

The script accepts a namespace and applies '- ../../../components/monitoring-rbac' line to 'components' field with a kustomize cli.

On success print nothing, and on failure print relevant error from kustomize cli. 

Notice that kustomize editing the file and uses specific annotation while listing the components, the '-' is at the first char and not the second on each line.

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **tumido** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @tumido` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fapps).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/apps/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["tumido"]} -->

tumido reviewed (commented):
/lgtm :rocket: :+1:

4n4nd reviewed (commented):
/lgtm

sesheta commented:
New changes are detected. LGTM label has been removed.

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **4n4nd** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fapps).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/apps/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["4n4nd"]} -->

sesheta commented:
@ipolonsk: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
aicoe-ci/prow/pre-commit | b5e4d6b0f4876794a1a7454018dd0480571d4172 | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--36f9ca3b-f5af-4432-bef8-af42508b403e/prow-logs/pr-logs/pull/operate-first_apps/572/pre-commit/1387092219269222400) | `/test pre-commit`

[Full PR test history](https://prow.operate-first.cloud/pr-history?org=operate-first&repo=apps&pr=572). [Your PR dashboard](https://prow.operate-first.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3Aipolonsk). Please help us and [open an issue](https://github.com/operate-first/apps/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ipolonsk commented:
Modified the script to be more consistent with the other scripts.
- Changed file name `enable_monitoring.sh` -> `enable-monitoring.sh`
- Changed PATH used from ''../$APP_NAME/.." -> ''$APP_NAME/.."
(So we can run it like the other scripts with `bash scripts/enable-monitoring.sh` directly from the main folder.)
- Changed the check on the number of arguments, now it gets only one argument. (other arguments will not be used)
- Added check if a namespace exists in base, prevent it from trying to navigate to a not existing folder. (return an error for the user to notice).
- Used function to wrap the commands. 

* since I changed the file name git shows this commit as a new file, sorry for the inconvenience.

larsks commented on scripts/enable-monitoring.sh:
```diff
@@ -0,0 +1,28 @@
+}
+
+if [ ! -d $APP_NAME/base/namespaces/$NAMESPACE ]; then
+	echo "Namespace '$NAMESPACE' is not exists in $APP_NAME/base/namespace/. Exiting."
```
Should be "does not exist" rather than "is not exists".

sesheta commented:
@ipolonsk: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
aicoe-ci/prow/pre-commit | 9efa4737a849c80b659e9a7e0ecd775ed34c44e4 | [link](https://prow.operate-first.cloud/view/s3/ci-prow-artifacts--36f9ca3b-f5af-4432-bef8-af42508b403e/prow-logs/pr-logs/pull/operate-first_apps/572/pre-commit/1387099799894888448) | `/test pre-commit`

[Full PR test history](https://prow.operate-first.cloud/pr-history?org=operate-first&repo=apps&pr=572). [Your PR dashboard](https://prow.operate-first.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3Aipolonsk). Please help us and [open an issue](https://github.com/operate-first/apps/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/apps/pull/572#pullrequestreview-646175056" title="Approved">larsks</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **4n4nd** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fapps).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/apps/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["4n4nd"]} -->

HumairAK commented:
awesome, can you also update the docs [here](https://github.com/operate-first/apps/blob/master/docs/odh/prometheus/add_service_monitoring.md#steps) (the first step) to use this script instead?

tumido reviewed (changes requested):
requesting changes for science! :thinking:

HumairAK commented:
can you squash these to 1 commit

tumido commented on scripts/enable-monitoring.sh:
```diff
@@ -0,0 +1,28 @@
+if [ $# -ne 1 ] || [ "$#" == "--help" ] || [ "$#" == "-h" ]; then
+    echo "Enable monitoring on a namespace"
+    echo -e "\nUsage: $0 NAMESPACE\n"
+	exit 0
```
can you please indent this the same?
![image](https://user-images.githubusercontent.com/7453394/116559372-aea3da00-a900-11eb-9eef-7f0d84858a28.png)

tumido reviewed (changes requested):
Please change this line `scripts/onboarding.sh` to `scripts`:

https://github.com/operate-first/apps/blob/999d2cacb8efe9f06c6018ae0461d0fda3724d6b/.pre-commit-config.yaml#L2

ipolonsk commented:
Hey all,
Sorry for some reason I pushed all the latest commits together with my last commit, working on fixing it.

tumido reviewed (approved):
/approve
/lgtm
Let's have one more pair of eyes. I like this! Great job @ipolonsk ! :slightly_smiling_face: 

/hold

sesheta commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/apps/pull/572#pullrequestreview-646175056" title="Approved">larsks</a>*, *<a href="https://github.com/operate-first/apps/pull/572#pullrequestreview-650436601" title="LGTM">tumido</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fapps).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/operate-first/apps/blob/master/OWNERS)~~ [tumido]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

sesheta commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/apps/pull/572#pullrequestreview-650448453" title="Approved">larsks</a>*, *<a href="https://github.com/operate-first/apps/pull/572#pullrequestreview-650436601" title="LGTM">tumido</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fapps).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/operate-first/apps/blob/master/OWNERS)~~ [tumido]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

HumairAK commented:
/unhold
/lgtm
